ssl/statem/statem_clnt.c

```diff
@@ -1136,6 +1136,21 @@ MSG_PROCESS_RETURN tls_process_server_hello(SSL *s, PACKET *pkt)
                session_id_len);
     }
 
+    if (!ssl_verify_client_session_version(s)) {
```

Comment:
(Ugh, my last comment was in reply to @indutny. GitHub's UI is so bad at concurrent conversations...)

@kroeckx Yeah, I mean that the version match check is made very similarly to the cipher match check in BoringSSL and that it's fatal. (I'd missed that it was fatal in the PR too. The `ssl_get_new_session` call threw me off and I can't read GitHub diffs. :-) )